modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -423,6 +425,7 @@ object Inputs {
       else if (arg.endsWith(".sc")) Right(Seq(Script(dir, subPath)))
       else if (arg.endsWith(".scala")) Right(Seq(ScalaFile(dir, subPath)))
       else if (arg.endsWith(".java")) Right(Seq(JavaFile(dir, subPath)))
+      else if (arg.endsWith(".c") || arg.endsWith(".h")) Right(Seq(CFile(dir, subPath)))
```

Comment:
Similarly, this will include `.c` and `.h` inputs passed explicitly as inputs, even for non-scala-native builds.
I think we should at least print a warning if someone passes such a source explicitly when not running Scala native (?)

Reply:
Good point, I kept thinking about it in the context of passing a directory (where unrecognized files are ignored, which lined up with what we are doing here), but I see there should be an error when unrecognized/irrelevant files are passed directly

Reply:
https://github.com/VirtusLab/scala-cli/issues/1346